Assets/_BuildingBlocks/Credentials/Scripts/KeyVault.cs

```diff
@@ -153,83 +146,79 @@ private IEnumerator TryAllSupportedAuthorizations(Uri uri, string username, stri
         {
             foreach (var supportedAuthorization in supportedAuthorizationTypes.Keys)
             {
-                var potentialAuth = CreateStoredAuthorization(supportedAuthorization, uri, passwordOrKey, username); // pass usernameOrKey before username for constructor order
-                yield return TrySupportedAuthorization(potentialAuth, uri);
-                if (storedAuthorizations.ContainsKey(potentialAuth.Domain)) yield break; // if the Auth test was succesful, stop looking.
-            }
-        }
-
-        private IEnumerator TrySupportedAuthorization(StoredAuthorization.StoredAuthorization potentialAuth, Uri uri)
-        {
-            var config = Config.Default();
-            config = potentialAuth.AddToConfig(config);
-            var request = Uxios.DefaultInstance.Get<byte[]>(uri, config); //get as byte[] so Uxios doesn't try to interpret the response
+                // pass usernameOrKey before username for constructor order
+                var potentialAuth = CreateStoredAuthorization(
+                    supportedAuthorization, uri, passwordOrKey, username
+                ); 
 
-            bool isAuthorized = false;
+                var promisedAuthorization = TrySupportedAuthorization(potentialAuth, uri) as Promise<IResponse>;
```

Comment:
Aangezien je uitleg van Uxios gisteren, betekent dit dus dat ze allemaal tegelijk worden verstuurd, terwijl het idee was om te wachten tot er een response komt totdat de volgende wordt verstuurd. Misschien moeten we dit in de toekomst een keer verbeteren, helemaal als een project heel veel remote layers bevat

Reply:
of toch misschien toch niet, hij wacht hieronder op het request dus dan stopt ook de foreach loop totdat die request voltooid is